pom.xml

```diff
@@ -131,25 +146,20 @@
             <plugin>
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
-                <version>2.3.7</version>
+                <version>3.2.0</version>
                 <extensions>true</extensions>
-                <executions>
-                    <execution>
-                        <id>bundle-manifest</id>
-                        <phase>process-classes</phase>
-                        <goals>
-                            <goal>manifest</goal>
-                        </goals>
-                    </execution>
-                </executions>
                 <configuration>
                     <supportedProjectTypes>
-                        <supportedProjectType>jar</supportedProjectType>
+                        <supportedProjectType>bundle</supportedProjectType>
                     </supportedProjectTypes>
                     <instructions>
-                        <!-- OSGi specific instruction -->
-                        <Import-Package>!io.airlift.command,!com.maxmind.geoip2,!com.google.api.client,*
-                        </Import-Package>
+                        <Import-Package>!Acme.Serve,!org.junit,!android.os,!android.util,!android.net.http,!com.google.appengine.api.urlfetch,!rx,!rx.subscriptions,*</Import-Package>
+                        <Brooklyn-Feature-Name>${project.name}</Brooklyn-Feature-Name>
+                        <Implementation-SHA-1>${buildNumber}</Implementation-SHA-1>
+                        <Implementation-Branch>${scmBranch}</Implementation-Branch>
+                        <!-- Embed-Transitive>true</Embed-Transitive -->
+                        <Embed-Dependency>*;scope=compile|runtime;groupId=!org.apache.brooklyn</Embed-Dependency>
```

Comment:
hi @andreaturli just to ask why it's necessary to use `Embed-Dependency` here, can this not be done with a Karaf feature rather than putting everything into an uber-jar?